src/coreComponents/mainInterface/initialization.cpp

```diff
@@ -21,7 +21,17 @@
 #include "mainInterface/GeosxVersion.hpp"
 
 // TPL includes
+#include <conduit/conduit.hpp>
+#include <hdf5.h>
+#include <metis.h>
 #include <optionparser.h>
+#include <parmetis.h>
+#include <superlu_defs.h>
+#include <umfpack.h>
+#include <vtkVersion.h>
```

Comment:
I'm not sure how I feel about a soup of all TPL includes in one place. For some of them, we've gone through some effort to restrict their inclusion to a single translation unit. Of course, this adds just one more unit so no big deal, and it's not like there are any actual conflicts between them presently. 

Some of them could also be handled by adding a `getVersionString()` function in their respective GEOSX "interfaces" (e.g. in `SuperLUDist.[hpp|cpp]`) that hide away the TPL includes and macros (still have to include the GEOSX-side headers here though). Others I guess don't have a single entry point in the code.

Reply:
I was about to comment the same thing (or so).
If this PR is merged as is, we'll end up with the main initialization depending on TPLs that have nothing to do here.
And this, regardless of protecting with `#ifdef/#endif` guards which is just a technical pirouette when we don't use it: when we do use the TPL, the issue is there.

`vtk` being a straightforward illustration since it should be invisible except where we're dealing with input and outputs.
If we adopt this pattern, we'll end up having the main interface depending on everything. And I expect that a lot of things depend on the main interface too...

> I'm not sure how I feel about a soup of all TPL includes in one place.

Personally I don't feel that's long term solution. For the dependency/architecture considerations more than for the translation units though.
Having neat and carefully trimmed dependencies is key to long term success.

To move forward, do we have other open-source software to get inspiration?
I expect some would generate some bootstrap tool to generate env info?

Reply:
I actually don't have too much of a problem with link-level dependencies, since `mainInterface` _by design_ is already dependent on everything else, and `initialization.cpp`'s role is specifically to initialize all dependencies. It seems if we want to have organized output of **runtime** dependency versions in one place, there's little chance to avoid this pattern. An alternative is to generate and inject a dependency version string at config time via CMake?

Reply:
I think it cmake of what is in this PR. Creating a getVersionString() for each dependency requires an interface wrapper for each dependency, and we still need to have the interface included in the `mainInterface`, so it really doesn't resolve anything I think?

@TotoGaz If you have a solution that you like, please feel free to suggest it. I added this PR because I thought we had a need, and I didn't think of anything better within reason and time constraints for implementing it. Perhaps the cmake solution would be the cleanest way I can think of.

Reply:
OK. We can use the cmake variables `PACKAGE_VERSION` created from the `find_package` call for: `Conduit, pugixml, RAJA, CHAI, Umpire, Adiak, Caliper, VTK, fmt, Python3, CUDAToolkit`. The rest are going to either involve parsing headers for their version info, or including like is done currently in this PR.

Reply:
> since mainInterface by design is already dependent on everything else, and initialization.cpp's role is specifically to initialize all dependencies.

When I read the code, I understand that the role is more to initialize the environment in which geosx is embedded.
Like `MPI`, the `logger`, etc. What's the "lingua franca" of GEOSX.
And in a quite agnostic way for the LAI (even though the implementation and the registration could be done in a more separate way: despite some details, `initialization` could be unaware of the `LAI` it's actually triggering).

In a similar way, couldn't the `LAI` provide it's own `getLAIVersion() -> string or map<string, string>` which can be displayed on screen? At least the current state of dependencies is not shifted.
I think we still need a solution for `metis`, `parmetis`, `MPI`, `HDF5`?
For `MPI`, I think a solution equivalent to the compilers' solutions is acceptable.

For `HDF5`, calling any of its executable outputs the version. e.g. 
```
# ./bin/h5ls -V
h5ls: Version 1.12.1
```
For `(par)metis` I don't know yet because their tools do not provide this kind of information.